sql/bloom_filters.h

```diff
@@ -28,18 +28,25 @@ SOFTWARE.
 #include <cmath>
 #include <vector>
 #include <algorithm>
+
+#define DEFAULT_IMPLEMENTATION
+#if __GNUC__ > 7
+#ifdef __x86_64__
 #ifdef HAVE_IMMINTRIN_H
 #include <immintrin.h>
-#if __GNUC__ > 7 && defined __x86_64__
+#undef DEFAULT_IMPLEMENTATION
 #define DEFAULT_IMPLEMENTATION    __attribute__ ((target ("default")))
 #define AVX2_IMPLEMENTATION __attribute__ ((target ("avx2,avx,fma")))
 #if __GNUC__ > 9
 #define AVX512_IMPLEMENTATION __attribute__ ((target ("avx512f,avx512bw")))
 #endif
 #endif
 #endif
-#ifndef DEFAULT_IMPLEMENTATION
-#define DEFAULT_IMPLEMENTATION
+#ifdef __aarch64__
+#include <arm_neon.h>
+#undef DEFAULT_IMPLEMENTATION
```

Comment:
I think a default implementation must always exist, that's why it's "default", for a case when CPU doesn't support necessary SIMD commands.

Reply:
Function multiversioning works only for x86, there's no `default` target on other platforms:
https://gcc.gnu.org/wiki/FunctionMultiVersioning
```
This support is available in GCC 4.8 and later. Support is only available in C++ for i386 targets.
```

It is under `#ifdef __aarch64__`, which guarantees NEON presence. It presumes ARMv8 at least. If we need to support this optimisation on earlier platforms (ARMv7?), it can be tricky.

There are also more advanced SVE/SVE2, but they don't seem to be widely available, e.g. there's none on our bb machine. And if they're available, they don't seem to offer registers larger than NEON does.